### ISSUE TITLE
Replace pyinstaller for OSX installation

### ISSUE THREAD

**Issue by cortesi** [mitmproxy/mitmproxy#394]
We need to move away from pyinstaller because:

- It causes weird behaviour in mitmdump, like failure to exit on keyboard interrupt
- It doesn't support Python3, which we're heading towards, and probably won't any time soon

I propose that we create mitmproxy installer script instead, which installs mitmproxy and all its dependencies in a virtualenv in the user's home directory, and then symlinks the executables in ~/bin.

This isolates us from the rest of the system, and might also form the basis of an installer for other platforms.

**Comment by mhils** [comment]
FWIW, Pyinstaller now has an experimental [Python 3 branch](http://pyinstaller.org).

**Comment by elitest** [comment]
I was watching some of the PyCon2015 videos last night, and watched [Olivier Grisel - Build and test wheel packages on Linux, OSX & Windows - PyCon 2015 ](https://www.youtube.com/watch?v=d-p6lJJObLU).  In what I have seen about Python packaging it seems that a lot of projects are going the wheel route.  The thing I don't like about wheel is that it really seems like a pip centered approach, and it doesn't seem like there is a good story there in terms of .msi/exe/dmg generation.  Having said that it does have good support for both 2.x/3.x.  Because this is a pure Python project I was able to generate a universal wheel in like 2 lines of code.

The video also discusses Travis a bit, and specifically a feature of it that I did not know existed.  Apparently it allows you to run builds on [OS X](http://docs.travis-ci.com/user/osx-ci-environment/) but it appears that they may be [at capacity for this](http://docs.travis-ci.com/user/multi-os/) at the moment.  

Also mentioned in the video is [Appveyor](http://www.appveyor.com/), also free for OSS, that is basically Travis for Windows.  Anyway just thought I would toss that out there.
